src/test/java/org/junitpioneer/testkit/assertion/TestCaseAssertBase.java

```diff
@@ -47,7 +48,7 @@
 	@Override
 	public TestCaseFailureAssert whichFailed() {
 		assertThat(actual.failed().count()).isEqualTo(1);
-		return this;
+		return new TestCaseAssertBase(actual.failed());
```

Comment:
Took me a while to understand this, but I think it makes sense and actually makes the code a bit clearer. Thanks!

Reply:
Sorry for unresolving this comment. I was unsure whether this change could lead to any incorrect usage (similar to how `CONTRIBUTING.md` shows `hasNumberOfFailedTests(3).hasSingleSucceededTest()`), but this does not seem to be an issue here. The returned `TestCaseFailureAssert` only allows performing assertions on the exception. Therefore I resolved this comment again.